### PR TITLE
PYTHON-1316 Remove Database.eval and database.SystemJS

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Depends on https://github.com/mongodb/mongo-python-driver/pull/540.

### CODE REVIEW

File: doc/migrate-to-pymongo4.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Removed :meth:`~pymongo.database.Database.eval` and
+:class:`~pymongo.database.SystemJS`. The eval command was deprecated in
+MongoDB 3.0 and removed in MongoDB 4.2.
+
+Code like this::
+
+  >>> result = database.eval('function (x) {return x;}', 3)
+
+can be changed to this with MongoDB <= 4.0::
```
Comment (behackett) [inline doc/migrate-to-pymongo4.rst:60]: You should say explicitly that there is no replacement for eval when using MongoDB 4.2+.

**Review by behackett — CHANGES_REQUESTED** [review]

**Review by prashantmital — COMMENTED** [review]
Does this not count as a removed feature with no migration path?

**Comment by ShaneHarvey** [comment]
> Does this not count as a removed feature with no migration path for MongoDB 4.2+? Should we add a line or two addressing this to that section?

I didn't really think it was a good fit for that category because `eval` will not work on MongoDB 4.2+ even on PyMongo 3.x.

**Review by prashantmital — APPROVED** [review]
LGTM

**Pull request merged**